src/hotspot/share/gc/z/zDirector.cpp

```diff
@@ -488,7 +488,7 @@ static bool rule_major_allocation_rate(const ZDirectorStats& stats) {
 
   // Calculate the GC cost for each reclaimed byte
   const double current_young_gc_time_per_bytes_freed = double(young_gc_time) / double(reclaimed_per_young_gc);
-  const double current_old_gc_time_per_bytes_freed = double(old_gc_time) / double(reclaimed_per_old_gc);
+  const double current_old_gc_time_per_bytes_freed = reclaimed_per_old_gc == 0 ? std::numeric_limits<double>::infinity() : double(old_gc_time) / double(reclaimed_per_old_gc);
```

Comment:
How about using some parentheses? To my understanding, the division has a higher precedence than the ternary conditional expression. See: https://en.cppreference.com/w/cpp/language/operator_precedence

Reply:
I do not mind parentheses. But ternary are the lowest precedence (if you do not count the `,` which I would almost always say is wrong to use without a surrounding `() / [] / {}`), so to me it seems superfluous.

Just to clarify the intent of this code is what we are getting with a higher precedence on division. That is:
```C
  const double current_old_gc_time_per_bytes_freed = ((reclaimed_per_old_gc == 0) ? (std::numeric_limits<double>::infinity()) : (double(old_gc_time) / double(reclaimed_per_old_gc)));
```

_Side Note:_
I also think I prefer immediately invoked lambdas when the ternaries get this long.
```C
  const double current_old_gc_time_per_bytes_freed = [&]() {
    if (reclaimed_per_old_gc == 0) {
      return std::numeric_limits<double>::infinity();
    }
    return double(old_gc_time) / double(reclaimed_per_old_gc);
  }();
```